next/app/components/common/GenerateButton/GenerateButtion.tsx

```diff
@@ -1,5 +1,22 @@
-import React from 'react'
+'use client'
+
+import React, { useState } from 'react'
+import s from './GenerateButton.module.css'
 
 export default function GenerateButton() {
-  return <></>
+  const [isLoading, setIsLoading] = useState(false)
+
+  const handleClick = () => {
+    setIsLoading(true)
+    // Simulate a loading process
+    setTimeout(() => {
+      setIsLoading(false)
+    }, 3000)
+  }
+
+  return (
+    <button className={s.generateButton} onClick={handleClick}>
+      {isLoading ? <div className={s.loader}></div> : <p>Generate</p>}
+    </button>
```

Comment:
ローディング中にボタンが押されないようにするため、disabled属性を付与しておきましょう！
```suggestion
    <button className={s.generateButton} onClick={handleClick} disabled={isLoading}>
      {isLoading ? <div className={s.loader}></div> : <p>Generate</p>}
    </button>
```